### PR TITLE
feat: open trade summary

### PR DESCRIPTION
**Pull request by chrisling-dev**
https://www.notion.so/prepo/FE-Project-Management-d3878c29e9534234a50f4f8838525bee?p=b152339d808040bdb22c6ef72c7d97c5&pm=s

Screenshot 2023-01-17 at 5 22 47 PM

https://user-images.githubusercontent.com/81092286/212891027-f769faca-7f4b-4703-bcdb-42e4cced2716.mov

### CODE REVIEW

**Review by chrisling-dev — COMMENTED** [review]
self reviewed

Screenshot 2023-01-17 at 5 22 47 PM

File: apps/frontend/core-dapp/src/features/trade/TradeStore.ts [hunk 1]
```diff
@@ -139,13 +140,53 @@ export class TradeStore {
...
+    const inRange =
+      this.tradingLongPriceAfterSlippage > lowerBound &&
+      this.tradingLongPriceAfterSlippage < upperBound
+
+    return inRange || this.openTradeAmountBN.eq(0)
+  }
+  // TODO: this is only used in trade summary modal which will be deleted soon
+  // delete this when we delete trade summary modal
+  get valuation(): { raw?: number | undefined; afterSlippage?: number | undefined } {
+    if (!this.selectedMarket || this.openTradeAmountOut === undefined) return {}
```
Comment (liqwiz) [inline apps/frontend/core-dapp/src/features/trade/TradeStore.ts:189]: just making sure you didn't miss the todo

**Review by liqwiz — APPROVED** [review]
approved with a comment

File: apps/frontend/core-dapp/src/features/trade/TradeTransactionSummary.tsx [hunk 2]
```diff
@@ -91,12 +93,22 @@ const TradeTransactionSummary: React.FC = () => {
...
+    if (selectedMarket === undefined) return 'Select a Market'
+    if (withinBound === false) return 'Insufficient Liquidity'
+    return undefined
+  }, [selectedMarket, withinBound])
+
   return (
     <TransactionSummary
-      overrideText={selectedMarket === undefined ? 'Select a Market' : undefined}
-      loading={openTradeUILoading(selectedMarket) || openTradeAmountBN === undefined}
+      overrideText={ovverrideButtonText}
```
Comment (XavierEkkel) [inline apps/frontend/core-dapp/src/features/trade/TradeTransactionSummary.tsx:104]:
```suggestion
      overrideText={overrideButtonText}
```

File: apps/frontend/core-dapp/src/features/trade/TradeTransactionSummary.tsx [hunk 3]
```diff
@@ -91,12 +93,22 @@ const TradeTransactionSummary: React.FC = () => {
     },
   ]
 
+  const ovverrideButtonText = useMemo(() => {
+    if (selectedMarket === undefined) return 'Select a Market'
+    if (withinBound === false) return 'Insufficient Liquidity'
```
Comment (XavierEkkel) [inline apps/frontend/core-dapp/src/features/trade/TradeTransactionSummary.tsx:98]:
```suggestion
    if (withinBound === false) return 'Unprofitable Trade'
```

File: apps/frontend/core-dapp/src/features/trade/TradeTransactionSummary.tsx [hunk 4]
```diff
@@ -28,6 +29,7 @@ const TradeTransactionSummary: React.FC = () => {
     selectedMarket,
     valuation: { raw, afterSlippage },
     direction,
+    withinBound,
```
Comment (XavierEkkel) [inline apps/frontend/core-dapp/src/features/trade/TradeTransactionSummary.tsx:32]:
```suggestion
    withinBounds,
```

File: apps/frontend/core-dapp/src/features/trade/TradeStore.ts [hunk 5]
```diff
@@ -139,13 +140,53 @@ export class TradeStore {
...
+    if (!valuationRange || !payoutRange) return undefined
+
+    return calculateValuation({
+      longTokenPrice: this.tradingLongPriceAfterSlippage,
+      payoutRange,
+      valuationRange,
+    })
+  }
+
+  get withinBound(): boolean | undefined {
```
Comment (XavierEkkel) [inline apps/frontend/core-dapp/src/features/trade/TradeStore.ts:168]:
```suggestion
  get withinBounds(): boolean | undefined {
```

**Review by XavierEkkel — APPROVED** [review]
LGTM after comments

File: apps/frontend/core-dapp/src/features/trade/TradeTransactionSummary.tsx [hunk 6]
```diff
@@ -91,12 +93,22 @@ const TradeTransactionSummary: React.FC = () => {
...
-      loading={openTradeUILoading(selectedMarket) || openTradeAmountBN === undefined}
+      overrideText={ovverrideButtonText}
+      loading={
+        openTradeUILoading(selectedMarket) ||
+        openTradeAmountBN === undefined ||
+        withinBound === undefined
+      }
       data={tradeTransactionSummary}
-      disabled={tradeDisabled || raw === undefined || selectedMarket === undefined}
+      disabled={tradeDisabled || raw === undefined || selectedMarket === undefined || !withinBound}
```
[the remark this answers is not in the document]
Comment (chrisling-dev) [reply apps/frontend/core-dapp/src/features/trade/TradeTransactionSummary.tsx:111]: nope it should be disabled both when it's loading and when it's false. disabled and loading doesn't conflict, we'd just look at withinBounds and think when should withinBounds make the button disabled? when it's loading and when it's not within bound (false)

**Review by XavierEkkel — APPROVED** [review]

**Pull request merged**